util/printchplenv

```diff
@@ -39,260 +109,278 @@ def print_mode(mode='list', anonymize=False):
+
+
+"""Filter CHPL_MAKE for --path (special case)"""
+def filter_path(chpl_env):
+    return chpl_env.name != 'CHPL_MAKE'
+
+
+"""Filter variables that are not selected in contents
+Requires a content argument via functools.partial
+"""
+def _filter_content(chpl_env, contents=None):
+    return chpl_env.content.intersection(contents)
+
+
+"""Return string to be printed for a given variable and print_format
+Requires a print_format argument
+"""
+def _print_var(env_var, value, print_format=None, shortname=None):
+    env = env_var.strip()
+    if print_format == 'pretty':
+        user_set_symbol = user_set(env)
+        return "{1}: {2}{0}\n".format(user_set_symbol, env_var, value)
+    elif print_format == 'simple':
+        return "{0}={1}\n".format(env, value)
+    elif print_format == 'make':
+        make_env_var = env_var.replace("CHPL_", "CHPL_MAKE_", 1).lstrip()
+        return "{0}={1}\n".format(make_env_var, value)
+    elif print_format == 'path':
+        if shortname:
+            ret = "{0}-{1}".format(shortname, value)
+        else:
+            ret = "{0}".format(value)
+        return ret + '/'
+    else:
+        raise ValueError("Invalid format '{0}'".format(print_format))
 
-    elif m == 'compiler' or m == 'runtime' or m == 'launcher':
-        stdout.write('\n')
 
+"""Return a string that contains the Chapel configuration variable info"""
+def printchplenv(contents, print_filters=None, print_format='pretty', recursive=False):
+    global CHPL_ENVS
 
+    if print_filters is None:
+        print_filters = ['tidy']
 
-def print_var(env_var, value, mode, short_name='', filters=None):
-    if mode == 'list' or mode == 'debug':
-        user_set = overrides.get_environ(env_var.strip(), '')
-        if user_set:
-          user_set = ' *'
-        else:
-          user_set = overrides.get_chplconfig(env_var.strip(), '')
-          if user_set:
-            user_set = ' +'
-        stdout.write("{1}: {2}{0}\n".format(user_set, env_var, value))
-    elif mode == 'make':
-        make_env_var = env_var.replace("CHPL_", "CHPL_MAKE_", 1).lstrip()
-        stdout.write("{0}={1}\n".format(make_env_var, value))
-    elif mode == 'simple':
-        stdout.write("{0}={1}\n".format(env_var.strip(), value))
-    elif mode == 'compiler' or mode == 'runtime' or mode == 'launcher':
-        if not filters or mode in filters:
-            if not print_var.first_time:
-                stdout.write("/")
-            else:
-                print_var.first_time = False
-
-            if not short_name or value.startswith(short_name):
-                stdout.write(value)
-            else:
-                stdout.write("{0}-{1}".format(short_name, value))
-    elif mode == 'shell':
-        stdout.write("export {0}='{1}'\n".format(env_var.strip(), value))
-    elif mode == 'cshell':
-        stdout.write("setenv {0} '{1}'\n".format(env_var.strip(), value))
-    else:
-        raise ValueError("Invalid mode '{0}'".format(mode))
+    if not recursive:
```

Comment:
I'm not really in to this 'recursive' argument. I think I'd do this instead:
 * call populate_values before calling printchplenv in main
 * remove the call to populate_values here
 * remove the 'recursive' argument
 * change populate_values argument from contents to compute_internal that is True or False
 * set compute_internal in main to 'internal' in contents

This reduces the complexity of the control flow and more clearly shows the current assumption that a recursive call within printchplenv cannot add internal to contents when it wasn't there before.

Reply:
Good call - I've opted for splitting `populate_values` into `compute_all_values()` and `compute_internal_values()`